cli/config.go

```diff
@@ -3,34 +3,84 @@ package cli
 import (
 	"errors"
 	"fmt"
-	"os"
 
+	"github.com/MakeNowJust/heredoc"
 	"github.com/odpf/compass/internal/client"
 	"github.com/odpf/compass/internal/server"
 	esStore "github.com/odpf/compass/internal/store/elasticsearch"
 	"github.com/odpf/compass/internal/store/postgres"
 	"github.com/odpf/compass/pkg/metrics"
 	"github.com/odpf/compass/pkg/statsd"
+	"github.com/odpf/salt/cmdx"
 	"github.com/odpf/salt/config"
 	"github.com/spf13/cobra"
-	"gopkg.in/yaml.v2"
 )
 
 const configFlag = "config"
 
-func cmdShowConfigs() *cobra.Command {
+var cliConfig *Config
+
+func configCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config <command>",
+		Short: "Manage server and client configurations",
+		Example: heredoc.Doc(`
+			$ compass config init
+			$ compass config list`),
+	}
+
+	cmd.AddCommand(configInitCommand())
+	cmd.AddCommand(configListCommand())
+
+	return cmd
+}
+
+func configInitCommand() *cobra.Command {
 	return &cobra.Command{
-		Use:   "configs",
-		Short: "Display configurations currently loaded",
-		Run: func(cmd *cobra.Command, args []string) {
-			cfg, err := loadConfig(cmd)
+		Use:   "init",
+		Short: "Initialize a new sevrer and client configuration",
+		Example: heredoc.Doc(`
+			$ compass config init
+		`),
+		Annotations: map[string]string{
+			"group": "core",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg := cmdx.SetConfig("compass")
+
+			if err := cfg.Init(&Config{}); err != nil {
+				return err
+			}
+
+			fmt.Printf("config created: %v\n", cfg.File())
+			return nil
+		},
+	}
+}
+
+func configListCommand() *cobra.Command {
```

Comment:
@sudo-suhas we can load the config from custom file location with a `-c` or `--config` flag.
So for loading the file from the current directory you can use `compass config list -c ./compass.yml` 

Otherwise, by default it will try to load the file which we made from config init command

Reply:
Edit: understood what you meant by current flow, will add the current directory to search for the config file, before looking for default location

Reply:
Can we please not break backward compatibility here? Not sure I see the value in doing so. Supporting a new default is okay but breaking existing functionality for insignificant gains should be avoided.

Reply:
> Edit: understood what you meant by current flow, will add the current directory to search for the config file, before looking for default location

if this is the approach we are taking, we are partially in agreement. Please look into the previous behaviour and ensure the same is supported.

Reply:
yes, I am adding the existing flow for loading  from root as well

Reply:
@sudo-suhas I have restored the option to load from the flags, or root. If files aren't available then I am loading from the file created from the init command defined in cmdx/config in salt. (Option 1)

An alternative option is to edit the cmdx package to find the file in the root. (Option 2) 
Or we can just copy the Loading part with configdir defined in here to compass (Option 3) https://github.com/odpf/salt/blob/main/cmdx/config.go#L83.

Let me know if the current changes are fine, or should we change it?